### PR TITLE
Refactor addition of chat and system messages to DOM

### PR DESCRIPTION
**Pull request by felipeolifre**

### CODE REVIEW

File: public/index.js [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+  } else if (message.type == 'system') {
+    listItem.textContent = message.body;
+    listItem.style.fontStyle = 'italic';
+  }
   document.getElementById('messages').appendChild(listItem);
 };
 
 document.addEventListener('DOMContentLoaded', () => {
   const socket = io();
+  let message = {};
```
Comment (olistic) [inline public/index.js]: Why did you initialize message up there? It's best to initialize variables as close to where they are used as possible.

File: public/index.js [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
 
-  socket.on('system message', message => {
-    const listItem = document.createElement('li');
-    listItem.style.fontStyle = 'italic';
-    listItem.textContent = message;
-    document.getElementById('messages').appendChild(listItem);
+  socket.on('system message', messageBody => {
```
Comment (olistic) [inline public/index.js]: You should set a standard on the message where all messages are objects, and every time you emit/listen to message events (chat and system) pass a message object.

Comment (olistic) [reply public/index.js]: And `messageBody` could be called just `body`, because it will live within the `message` object.

File: public/index.js [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
+    message.type = 'system';
+    message.body = messageBody;
+    addMessageToDOM(message);
   });
 
-  socket.on('chat message', ({ nickname, message }) => {
-    addChatMessageToDOM(nickname, message);
+  socket.on('chat message', ({ nickname, messageBody }) => {
```
Comment (olistic) [inline public/index.js]: Idem regarding `messageBody`, you can just call it `body` and it will be fine IMO.

**Comment by felipeolifre** [comment]
Thanks for your comments @olistic - It looks much better now :)

**Pull request merged**